lib/foreman/renderer/scope/macros/base.rb

```diff
@@ -11,36 +12,85 @@ module Base
 
           delegate :medium_uri, to: :medium_provider
 
+          apipie :class, 'Base macros to use within a template' do
+            name 'Base'
+            sections only: %w[all reports provisioning jobs]
+          end
+
+          apipie :method, 'Returns true if subnet has a given parameter set, false otherwise' do
+            desc 'This does not take inheritance into consideration, it only
+              searches for parameters assigned directly to the subnet.'
+            required :subnet, 'Subnet', desc: 'a __Subnet__ object for which we check the parameter presence'
+            required :param_name, String, desc: 'a parameter __name__ to check the presence of'
+            returns one_of: [true, false]
+            raises error: WrongSubnetError, desc: 'when user passes non-subnet object as __subnet__ parameter'
+          end
           def subnet_has_param?(subnet, param_name)
             validate_subnet(subnet)
             subnet.parameters.exists?(name: param_name)
           end
 
+          apipie :method, 'Returns the value of global setting' do
+            required :name, String, desc: 'the name of the setting which can be found by hovering the setting with mouse cursor in the UI, or via API/CLI'
+            optional :blank_default, Object, desc: 'if the setting is not set to any value, this value will be returned instead', default: nil
+            returns Object, desc: 'The value of global setting, e.g. String, Integer, Array, Provisioning Template etc'
+            example "global_setting('foreman_url') # => 'https://foreman.example.com'"
+            example "global_setting('outofsync_interval', 30) # => 30"
+          end
```

Comment:
Actually an important fact is that not all settings are exposed, only those which are allowed via `Foreman::Renderer.config.allowed_global_settings`.

Reply:
Can't you simply list them in the documentation? It is just an array I think, append it.

Reply:
@lzap, sorry didn't notice your comment. Added list of the allowed settings.